docs/source/tutorials/cascaded_architecture.rst

```diff
@@ -0,0 +1,154 @@
+some of the key parameters to use cascaded models.
+
+- ``object_detection_params:object_detection_path``: Location of the object detection model. This parameter corresponds
+  to the prerequisite model path  to the trained model (see :ref:`Prerequisite`). This spinal cord segmentation model
+  will be applied to the images and a bounding box will be created around this mask to crop the image.
+
+  .. code-block:: xml
+
+     "object_detection_path": "<SPINAL_CORD_SEG_LOG_DIRECTORY>/spineGeneric/seg_sc_t1_t2_t2s_mt"
+
+- ``object_detection_params:safety_factor``: Multiplicative factor to apply to each dimension of the bounding box. To
+  ensure all the CSF is included, a safety factor should be applied to the bounding box generated from the spinal cord.
+  A safety factor of 200% on each dimension is applied on the height and width of the image.
+
+  .. code-block:: xml
+
+     "safety_factor": [2, 2, 1]
+
+- ``loader_parameters:target_suffix``: Suffix of the ground truth segmentation. The ground truth are located under the
+  ``DATASET/derivatives/labels`` folder. The suffix for CSF labels in this dataset is ``_csfseg-manual``:
+
+  .. code-block:: xml
+
+     "target_suffix": ["_csfseg-manual"]
+
+- ``loader_parameters:contrast_params``: Contrast(s) of interest. The CSF labels are only available in T2w contrast in
+  the example dataset.
+
+  .. code-block:: xml
+
+     "contrast_params": {
+         "training_validation": ["T2w"],
+         "testing": ["T2w"],
+         "balance": {}
+     }
+
+- ``transformation:CenterCrop:size``: Crop size in pixel. Images will be cropped or padded to fit these dimensions. This
+  allows all the images to have the same size during training. Since the images will be cropped around the spinal cord,
+  the image size can be reduced. The ``preprocessing`` parameter indicates this transformation will only be applied
+  once at the beginning of the training.
+
+  .. code-block:: xml
+
+     "CenterCrop": {
+         "size": [64, 64],
+         "preprocessing": true
+     }
+
+Train model
+-----------
+
+Once the configuration file is ready, run the training. `ivadomed` has an option to find optimal threshold to binarize
+the test data to find the best balance between false positives and false negatives. Add the flag `-t` with an increment
+between 0 and 1 to get best threshold on the validation set (e.i. -t 0.1 will return the best threshold between 0.1,
+0.2, ..., 0.9)
+
+.. code-block:: bash
+
+   ivadomed -c config.json -t 0.01
+
+.. note::
+
+   If a `compatible GPU <https://pytorch.org/get-started/locally/>`_ is available, it will be used by default. Otherwise, training will use the CPU, which will take
+   a prohibitively long computational time (several hours).
+
+At the end of the training, the optimal threshold will be indicated:
+
+.. code-block:: console
+
+   Running threshold analysis to find optimal threshold
+	Optimal threshold: 0.01
+	Saving plot: spineGeneric/roc.png
+
```

Comment:
```suggestion
... and automatically saved in the configuration file to be used when launching the testing command.
```

Reply:
I find this confusing since the threshold is saved only in the package model. If I read this, I believe the threshold will automatically change in my current config file (the one I am using) which is not the case. Side note, but I think if the user uses the `-t` flag it is safe to directly change the config file he is using and change it to the right threshold, what do you think?

Reply:
You are very right! I was thinking of the `automate_training` were we dont need to do this. But in normal cases we do need to change it manually. All good as you did :-)